### PR TITLE
docs: add additional info on use_legacy_flow_control parameter

### LINKED ISSUE #300: Incomplete docstrings for the use_legacy_flow_control parameter
#231 added the `use_legacy_flow_control` parameter to subscriber client and documented it, but it did not document its type and and the default value in all places ([example](https://googleapis.dev/python/pubsub/latest/subscriber/api/client.html#google.cloud.pubsub_v1.subscriber.client.Client.subscribe) - missing from the parameter list). Docstrings should be updated accordingly.

### PR DESCRIPTION
**Pull request by plamut**
Closes #300.

This PR adds missing info on `use_legacy_flow_control` parameter where applicable

PR checklist:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-pubsub/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py [hunk 1]
```diff
@@ -106,9 +106,11 @@ class StreamingPullManager(object):
             ``projects/{project}/subscriptions/{subscription}``.
         flow_control (~google.cloud.pubsub_v1.types.FlowControl): The flow
             control settings.
-        use_legacy_flow_control (bool): Disables enforcing flow control settings
-            at the Cloud PubSub server and uses the less accurate method of only
-            enforcing flow control at the client side.
+        use_legacy_flow_control (bool):
+            If set to ``True``, it disables enforcing flow control settings on the
+            Cloud PubSub server. Defaults to ``False``.
```
Comment (anguillanneuf) [inline google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py:111]:
```suggestion
            Cloud Pub/Sub server. Defaults to ``False``.
```

File: google/cloud/pubsub_v1/subscriber/client.py [hunk 2]
```diff
@@ -199,6 +199,11 @@ def callback(message):
                 *scheduler* to use when executing the callback. This controls
                 how callbacks are executed concurrently. This object must not be shared
                 across multiple SubscriberClients.
+            use_legacy_flow_control (bool):
+                If set to ``True``, it disables enforcing flow control settings on the
+                Cloud PubSub server. Defaults to ``False``.
```
Comment (anguillanneuf) [inline google/cloud/pubsub_v1/subscriber/client.py:204]:
```suggestion
                Cloud Pub/Sub server. Defaults to ``False``.
```

**Review by anguillanneuf — APPROVED** [review]

File: google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py [hunk 3]
```diff
@@ -106,9 +106,11 @@ class StreamingPullManager(object):
             ``projects/{project}/subscriptions/{subscription}``.
         flow_control (~google.cloud.pubsub_v1.types.FlowControl): The flow
             control settings.
-        use_legacy_flow_control (bool): Disables enforcing flow control settings
-            at the Cloud PubSub server and uses the less accurate method of only
-            enforcing flow control at the client side.
+        use_legacy_flow_control (bool):
+            If set to ``True``, it disables enforcing flow control settings on the
+            Cloud Pub/Sub server. Defaults to ``False``.
+            The less accurate method of enforcing flow control on the client side
```
Comment (kamalaboulhosn) [inline google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py:112]: I don't know if I'd call it "less accurate." The flow control limits will still be enforced the same way with regard to the user callback. The difference is that there will be fewer messages buffered in the client library itself.

Comment (plamut) [reply google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py:112]: The "less accurate" bit was added [here](https://github.com/googleapis/python-pubsub/pull/231/files), but it's indeed a bit odd - being less accurate would imply that there are bugs in the client flow control logic. I'll remove that.

File: google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py [hunk 4]
```diff
@@ -106,9 +106,11 @@ class StreamingPullManager(object):
             ``projects/{project}/subscriptions/{subscription}``.
         flow_control (~google.cloud.pubsub_v1.types.FlowControl): The flow
             control settings.
-        use_legacy_flow_control (bool): Disables enforcing flow control settings
-            at the Cloud PubSub server and uses the less accurate method of only
-            enforcing flow control at the client side.
+        use_legacy_flow_control (bool):
```
Comment (pradn) [inline google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py:109]: Perhaps? I prefer a bit of repetition with parallel construction to make things extra clear.

If set to ``True``, flow control at the Cloud Pub/Sub server is disabled, though client-side flow control is still enabled. If set to `False``, both server-side and client-side flow control are enabled.

Comment (plamut) [reply google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py:109]: Reworded, more clarity is always good.

**Comment by plamut** [comment]
@pradn Ping, we just need one final rubber stamp, thanks!

**Review by hongalex — APPROVED** [review]

**Pull request merged**